extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusSecurityIdentity.java

```diff
@@ -149,6 +162,7 @@ public static class Builder {
         Set<Permission> permissions = new HashSet<>();
         Map<String, Object> attributes = new HashMap<>();
         List<Function<Permission, Uni<Boolean>>> permissionCheckers = new ArrayList<>();
+        List<BiFunction<SecurityIdentity, Permission, Uni<Boolean>>> permissionCheckersWithIdentity = new ArrayList<>();
```

Comment:
I'm not sure putting it here is great, this builder is really for the users who write the augmentation, IMHO the original code where the permission checker is added after the identity was created was fine

Reply:
I usually don't use context suggestions for methods that are not available (like private methods, or package-private when outside of the package) in my IDE. That is why I don't think it can confuse users. I only use Intellij Idea though, maybe eclipse is not that smart?

Sorry if it is too obvious, but let me put into context why I have done it - it allows you to propagate identity attributes without building identity twice inside one augmentor.

in any case, I agree it is not super nice, I can drop it if you deem it necessary.

Reply:
also, I'd much prefer to make RoutingContext injectable as method parameter. The reason why I didn't do it is that it is very WS Next http upgrade specific. in any other case I am aware of you already have `RoutingContext` injectable from CDI.

Reply:
I'll create  a new class and avoid adding a new method to `QuarkusSecurityIdentity` builder in order to unblock this PR. I'll push it in an hour.

Reply:
Sorry Michal, but indeed, it will most likely look nicer, IMHO the builder should only offer code/methods that can be of of help when users work with this builder in their custom augmentors

Reply:
np, done